### PR TITLE
Use grpc_core::RefCount for grpc_call and mark Unref path unlikely.

### PR DESCRIPTION
**Pull request by soheilhy**
Unfortunately, we cannot use RefCount for `batch_control`, unless we support
reset API. So, for now, I simply marked its unref path as unlikely.

### CODE REVIEW

File: src/core/lib/surface/call.cc [hunk 1]
```diff
@@ -553,10 +553,10 @@ static void destroy_call(void* call, grpc_error* error) {
                                             grpc_schedule_on_exec_ctx));
 }
 
-void grpc_call_ref(grpc_call* c) { gpr_ref(&c->ext_ref); }
+void grpc_call_ref(grpc_call* c) { c->ext_ref.Ref(); }
 
 void grpc_call_unref(grpc_call* c) {
-  if (!gpr_unref(&c->ext_ref)) return;
+  if (GPR_UNLIKELY(!c->ext_ref.Unref())) return;
```
Comment (AspirinSJL) [inline src/core/lib/surface/call.cc]: How is the likelihood determined? I guess it's actually likely that `c->ext_ref.Unref() == false`?

Comment (soheilhy) [reply src/core/lib/surface/call.cc]: Oh darn. Thanks for catching this . This is a typo. My brain said likely but my hand wrote unlikely. I'm away from desk, will fix tonight.

Comment (soheilhy) [reply src/core/lib/surface/call.cc]: This should be fixed now. Thank you for catching this again.

**Comment by soheilhy** [comment]
Thank you!

**Comment by soheilhy** [comment]
Both interop tool failures.

**Pull request merged**